src/p2p_async.rs

```diff
@@ -58,18 +56,22 @@ impl<T: AsyncRead + AsyncWrite + Unpin + Send> P2psConnAsync<T> {
 
         // listen for response with their public key
         let mut buffer = [0u8; 32];
-        stream.read(&mut buffer).await?;
+        stream.read_exact(&mut buffer).await?;
 
         // generate encryption key with private key and their public key
-        let key = keys.generate_encryption_key(&Keys::public_key_from_bytes(buffer)?);
+        let key = keys.generate_encryption_key(&Keys::public_key_from_bytes(buffer)?)?;
```

Comment:
Since no error is going to be thrown from **public_key_from_bytes**, there is no need to expect an error
```suggestion
        let key = keys.generate_encryption_key(&Keys::public_key_from_bytes(buffer))?;
```